components/search/src/selector.rs

```diff
@@ -317,4 +344,248 @@ mod tests {
+        );
+
+        result = Arc::clone(&selector).filter_engine_configuration(SearchUserEnvironment {
+            locale: "fi".into(),
+            region: "FR".into(),
+            update_channel: SearchUpdateChannel::Default,
+            distribution_id: String::new(),
+            experiment: String::new(),
+            app_name: SearchApplicationName::FocusIos,
+            version: String::new(),
+        });
+
+        assert!(result.is_ok(), "Should not have errored: `{result:?}`");
+        assert_eq!(
+            result.unwrap(),
+            RefinedSearchConfig {
+                engines: vec!(
+                    SearchEngineDefinition {
+                        aliases: Vec::new(),
+                        charset: "UTF-8".to_string(),
+                        classification: SearchEngineClassification::General,
+                        identifier: "test1".to_string(),
+                        name: "Test 1".to_string(),
+                        order_hint: None,
+                        partner_code: String::new(),
+                        telemetry_suffix: None,
+                        urls: SearchEngineUrls {
+                            search: SearchEngineUrl {
+                                base: "https://example.com/1".to_string(),
+                                method: "GET".to_string(),
+                                params: Vec::new(),
+                                search_term_param_name: Some("q".to_string())
+                            },
+                            suggestions: None,
+                            trending: None
+                        }
+                    },
+                    SearchEngineDefinition {
+                        aliases: Vec::new(),
+                        charset: "UTF-8".to_string(),
+                        classification: SearchEngineClassification::General,
+                        identifier: "test2".to_string(),
+                        name: "Test 2".to_string(),
+                        order_hint: None,
+                        partner_code: String::new(),
+                        telemetry_suffix: None,
+                        urls: SearchEngineUrls {
+                            search: SearchEngineUrl {
+                                base: "https://example.com/2".to_string(),
+                                method: "GET".to_string(),
+                                params: Vec::new(),
+                                search_term_param_name: Some("search".to_string())
+                            },
+                            suggestions: None,
+                            trending: None
+                        }
+                    },
+                ),
+                app_default_engine_id: "test1".to_string(),
+                app_default_private_engine_id: None
+            }, "Should have selected the single engine, as the environments do not match for the other two"
```

Comment:
This comment doesn't seem to match the selection. It looks like there's two selected `searchEngineDefintions` that are being asserted for. 

What I'm understanding is: 
- It selected engine `test1` because that's applied to all applications.
- It also selected engine `test2` because of "applications": ["firefox-android", "focus-ios"], which matches `app_name: SearchApplicationName::FocusIos`

Reply:
Oh this was a copy & paste error. I'll improve it.